src/core/lispList.cc

```diff
@@ -287,16 +290,31 @@ CL_DEFUN T_sp cl__listSTAR(VaList_sp vargs) {
   return result.cons();
 }
 
+// Should verify the type of the argument
+// Theoretically it could be a bignum
+// But there are not bignum length list (there is not enough memory)
+// so for a Fixnum we do the real test
+// and for a Bignum we test if it is positive and than return the list
 
 CL_LAMBDA(list &optional (on 1));
 CL_DECLARE();
 CL_DOCSTRING("last - see CLHS");
-CL_DEFUN T_sp cl__last(T_sp list, int n) {
+CL_DEFUN List_sp cl__last(List_sp list, Integer_sp in) {
   if (list.nilp())
     return list;
-  if (n < 0)
-    CELL_ERROR(make_fixnum(n));
-  if (Cons_sp clist = list.asOrNull<Cons_O>()) {
+  if (core__bignump(in)) {
```

Comment:
Could you rearrange the code so that the first test is:  if (in.fixnump()) { ... fixnum case ...} else if (core__bignump(in)) { ... bignum case ... }.

The fixnum test is very fast (it checks the tag) and it will be the much more common case.